### PR TITLE
PR for #2174: Upgrade eslint-plugin-react and fix eslint errors

### LINKED ISSUE #2174: CI: Updating eslint-plugin-react@7.20.0 will yield new lint errors
### Q&A (please complete the following information)
 - OS: macOS
 - Browser: n/a
 - Version: na/
 - Method of installation: npm
 - Swagger-Editor version: 3.11.1
 - Swagger/OpenAPI version: n/a

### Content & configuration

 - Node 13 (likely irrelevant)
 - `"eslint-plugin-react": "^7.20.0",`

### Describe the bug you're encountering

The following errors does not currently result when using `eslint-plugin-react@4.3.0`

```
Warning: React version not specified in eslint-plugin-react settings. See https://github.com/yannickcr/eslint-plugin-react#configuration .

~/swagger-editor/src/plugins/editor/components/editor.jsx
  243:5  error  componentWillMount is deprecated since React 16.9.0, use UNSAFE_componentWillMount instead, see https://reactjs.org/docs/react-component.html#unsafe_componentwillmount. Use https://github.com/reactjs/react-codemod#rename-unsafe-lifecycles to automatically update your components                       react/no-deprecated
  262:5  error  componentWillReceiveProps is deprecated since React 16.9.0, use UNSAFE_componentWillReceiveProps instead, see https://reactjs.org/docs/react-component.html#unsafe_componentwillreceiveprops. Use https://github.com/reactjs/react-codemod#rename-unsafe-lifecycles to automatically update your components  react/no-deprecated

~/swagger-editor/src/standalone/topbar/DropdownMenu.js
   60:23  error  Do not use findDOMNode  react/no-find-dom-node
   98:18  error  Do not use findDOMNode  react/no-find-dom-node
  118:19  error  Do not use findDOMNode  react/no-find-dom-node

~/swagger-editor/src/standalone/topbar-menu-edit-convert/components/convert-modal.jsx
  105:23  error  `'` can be escaped with `&apos;`, `&lsquo;`, `&#39;`, `&rsquo;`  react/no-unescaped-entities
  172:13  error  `'` can be escaped with `&apos;`, `&lsquo;`, `&#39;`, `&rsquo;`  react/no-unescaped-entities

✖ 7 problems (7 errors, 0 warnings)
```

### To reproduce...

```js
npm i eslint-plugin-react@7.20.0
npm run lint-errors
```

### Expected behavior

When updating to `eslint-plugin-react@7.x`, also fix the above errors

### PR DESCRIPTION
**Pull request by matthewgallo**
### Description
I upgraded the `eslint-plugin-react` package and made sure to fix all of the outstanding eslint errors. I was able to confirm by running `npm run lint` that eslint returned with 0 errors.

### Motivation and Context
Fixes #2174 

### How Has This Been Tested?
I ran `npm run lint` to make sure that eslint returned with 0 errors.

### My PR contains... 

- [ ] No code changes (`src/` is unmodified: changes to documentation, CI, metadata, etc.)
- [X] Dependency changes (any modification to dependencies in `package.json`)
- [X] Bug fixes (non-breaking change which fixes an issue)
- [ ] Improvements (misc. changes to existing features)
- [ ] Features (non-breaking change which adds functionality)

### My changes...
- [ ] are breaking changes to a public API (config options, System API, major UI change, etc).
- [ ] are breaking changes to a private API (Redux, component props, utility functions, etc.).
- [ ] are breaking changes to a developer API (npm script behavior changes, new dev system dependencies, etc).
- [X] are not breaking changes.

### Documentation
- [ ] My changes do not require a change to the project documentation.
- [ ] My changes require a change to the project documentation.
- [ ] If yes to above: I have updated the documentation accordingly.

### Automated tests
- [X] My changes can not or do not need to be tested.
- [ ] My changes can and should be tested by unit and/or integration tests.
- [ ] If yes to above: I have added tests to cover my changes.
- [ ] If yes to above: I have taken care to cover edge cases in my tests.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by tim-lai** [comment]
@matthewgallo Thanks for the PR! Is eslint complaining about `UNSAFE_componentWillMount`, which is a React 16 method?

**Comment by matthewgallo** [comment]
Sure! It looks like `eslint` is only giving a warning on those methods. When I run `yarn lint` I only see warnings listed now.
```
243:5  warning  Identifier 'UNSAFE_componentWillMount' is not in camel case         camelcase
262:5  warning  Identifier 'UNSAFE_componentWillReceiveProps' is not in camel case  camelcase
```

File: src/plugins/editor/components/editor.jsx [hunk 1]
```diff
@@ -240,7 +240,7 @@ export default function makeEditor({ editorPluginsToRun }) {
       editor.setReadOnly(readOnly)
     }
 
-    componentWillMount() {
+    UNSAFE_componentWillMount() {
```
Comment (tim-lai) [inline src/plugins/editor/components/editor.jsx:243]: should not use React 16 method here.

File: src/plugins/editor/components/editor.jsx [hunk 2]
```diff
@@ -259,7 +259,7 @@ export default function makeEditor({ editorPluginsToRun }) {
       win.document.removeEventListener("click", this.onClick)
     }
 
-    componentWillReceiveProps(nextProps) {
+    UNSAFE_componentWillReceiveProps(nextProps) {
```
Comment (tim-lai) [inline src/plugins/editor/components/editor.jsx:262]: should not use React 16 method here.

**Review by tim-lai — CHANGES_REQUESTED** [review]

File: src/plugins/editor/components/editor.jsx (hunk 1 above)
Comment (matthewgallo) [reply src/plugins/editor/components/editor.jsx:243]: I supposed we could disable eslint for these two lines that are using the deprecated lifecycle methods or I could find a way to use a different lifecycle method that will work without any eslint errors. What do you think @tim-lai?

Comment (matthewgallo) [reply src/plugins/editor/components/editor.jsx:243]: Ideally, we shouldn't disable if we don't need to so I'll start off with refactoring these lifecycle methods.

Comment (tim-lai) [reply src/plugins/editor/components/editor.jsx:243]: for `componentWillMount`, it is safe to move to `componentDidMount`.

File: src/plugins/editor/components/editor.jsx (hunk 2 above)
Comment (tim-lai) [reply src/plugins/editor/components/editor.jsx:262]: `componentWillReceiveProps` is trickier, as I think there might be underlying performance reasons for it (ace, large api definitions). I think it is safer for now to explicitly ignore this warning in eslint.

Comment (matthewgallo) [reply src/plugins/editor/components/editor.jsx]: Sure, indeed it does seem like this method is a bit more involved. I've fixed this now

File: src/plugins/editor/components/editor.jsx (hunk 1 above)
Comment (matthewgallo) [reply src/plugins/editor/components/editor.jsx]: Fixed

**Review by tim-lai — APPROVED** [review]

**Comment by tim-lai** [comment]
@matthewgallo Merged! Thanks for the contribution!

**Pull request merged**